firecrown/cosmosis/run.py

```diff
@@ -0,0 +1,207 @@
+    values = _make_cosmosis_values(config['parameters'])
+    pool = _make_parallel_pool(config['sampler'])
+    pipeline = _make_cosmosis_pipeline(data, values, pool)
+
+    # Actually run the thing
+    run_cosmosis(None, pool=pool, ini=ini, pipeline=pipeline, values=values)
+
+    if pool is not None:
+        pool.close()
+
+
+def _make_parallel_pool(config):
+    """ Set up a parallel process pool.
+
+    Parameters
+    ----------
+
+    config: dict
+        Sampler configuration section of the input
+
+    Will look for the 'mpi' command in the parallel section
+
+    Returns
+    -------
+
+    pool: CosmoSIS MPIPool object
+        parallel process pool
+    """
+
+    from cosmosis.runtime.mpi_pool import MPIPool
+
+    # There is a reason to make the user actively
+    # request to use MPI rather than just checking -
+    # on many systems, including, importantly, NERSC,
+    # trying to import MPI when not running under the
+    # MPI environment will cause a crash
+    use_mpi = config.get('mpi', False)
+
+    if use_mpi:
+        pool = MPIPool()
+        if pool.size == 1:
+            print("Have mpi set to True but one process.")
+            print("I will ignore and run in serial mode.")
+            pool = None
+    else:
+        pool = None
+        print("Running in serial mode (one process).")
+
+    return pool
+
+
+def _make_cosmosis_pipeline(data, values, pool):
+    """ Build a CosmoSIS pipeline.
+
+    Parameters
+    ----------
+
+    values: Inifile
+        Cosmosis object representing the input parameter values
+
+    Returns
+    -------
+
+    pipeline: CosmoSIS pipeline objects
+        Instantiated pipeline but with no modules
+    """
+    from cosmosis.runtime.pipeline import LikelihoodPipeline
+    from cosmosis.runtime.module import Module
+    from cosmosis.runtime.utils import stdout_redirected
+
+    # Lie to CosmoSIS about where it is installed.
+    os.environ['COSMOSIS_SRC_DIR'] = '.'
+
+    # Build the pipeline that evaluates the likelihood.
+    # We avoid printing various bits of output info by silencing stdout on
+    # worker nodes.
+    if (pool is None) or pool.is_master():
+        pipeline = LikelihoodPipeline(load=False, values=values,
+                                      priors=None)
+    else:
+        with stdout_redirected():
+            pipeline = LikelihoodPipeline(load=False, values=values,
+                                          priors=None)
+
+    sys.stdout.flush()
+    # Set up a single cosmosis module, which will be the interface
+    # file in the same directory as this one
+    module = Module('firecrown', COSMOSIS_INTERFACE)
+    module.setup_functions(data)
+    pipeline.modules = [module]
+
+    return pipeline
+
+
+def _make_cosmosis_config(config):
+    """ Extract a cosmosis configuration object from a config dict
+
+    Parameters
+    ----------
+    config: dict
+        Configuration dictionary of 'sampler' section
```

Comment:
Ditto on the name of the parameter.

Reply:
Not quite sure what you mean here?

Reply:
It appears the function wants the sampler section of the config. Thus calling the parameter sampler should give people a strong hint as to what is being handled in the function.